shell/components/form/ResourceTabs/composable.ts

```diff
@@ -39,7 +39,7 @@ export const useTabCountWatcher = () => {
 
 export const useTabCountUpdater = () => {
   const tabKey = randomStr();
-  const updateCount = inject<UpdateCountFn>(UPDATE_COUNT_PROVIDER_KEY);
+  const updateCount = inject<UpdateCountFn | undefined>(UPDATE_COUNT_PROVIDER_KEY, undefined);
```

Comment:
This doesn't fix the noted issue but it does resolve a warning I was seeing in the console when a table was rendered outside of a tab.

<img width="630" height="472" alt="image" src="https://github.com/user-attachments/assets/ea29c6a8-6dad-4089-be0f-6ecefc964b4b" />